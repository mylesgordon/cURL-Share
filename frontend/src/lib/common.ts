export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}