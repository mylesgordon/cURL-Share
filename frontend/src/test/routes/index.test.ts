import Index from '../../routes/+page.svelte';
import { render } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';

describe('Index page', () => {
    testMatchingSnapshot(Index)

	it('should display the correct title', () => {
		render(Index);
		expect(document.title).toEqual('Home | cURL Share');
	});
});
