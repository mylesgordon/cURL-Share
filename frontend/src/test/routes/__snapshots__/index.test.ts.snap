// Vitest Snapshot v1

exports[`Index page > should match snapshot 1`] = `
<body>
  <div>
    <!--&lt;Meta&gt;-->
     
    <div
      class="flex flex-row gap-2 justify-between items-center"
    >
      <div
        class="flex gap-2 items-center"
      >
        <h2>
          Projects
        </h2>
         
      </div>
       
      <div
        class="flex s-MMN3rRE-BmEv"
      >
        <div
          class="w-100 s-JZthjkXlZrq3"
        >
          <label
            class="label screenreader-only s-JZthjkXlZrq3"
            for=""
          >
            
          </label>
           
          <input
            class="input s-JZthjkXlZrq3"
            id=""
            placeholder="Search..."
            type="text"
          />
           
        </div>
        <!--&lt;Input&gt;-->
         
      </div>
      <!--&lt;Search&gt;-->
    </div>
     
    
    <div
      class="card card-shadow card-rounded card-border mt-4 s-CTyDWs24qi3Z"
    >
      <a
        class="p-4 w-48 truncate"
        data-testid="project-list-entry-1-title"
        href="/project/1"
      >
        the cool project
      </a>
       
      <p
        class="p-4 flex-grow flex-shrink"
        data-testid="project-list-entry-1-description"
      >
        this is a project
      </p>
    </div>
    <!--&lt;Card&gt;-->
    <!--&lt;ProjectListEntry&gt;-->
    
    <div
      class="card card-shadow card-rounded card-border mt-4 s-CTyDWs24qi3Z"
    >
      <a
        class="p-4 w-48 truncate"
        data-testid="project-list-entry-0-title"
        href="/project/0"
      >
        the other project
      </a>
       
      <p
        class="p-4 flex-grow flex-shrink"
        data-testid="project-list-entry-0-description"
      >
        this is another project
      </p>
    </div>
    <!--&lt;Card&gt;-->
    <!--&lt;ProjectListEntry&gt;-->
    
    
    <!--&lt;+page&gt;-->
  </div>
</body>
`;

exports[`Index page > should match snapshot 2`] = `
<body>
  <div>
    <!--&lt;Meta&gt;-->
     
    <div
      class="flex flex-row gap-2 justify-between items-center"
    >
      <div
        class="flex gap-2 items-center"
      >
        <h2>
          Projects
        </h2>
         
      </div>
       
      <div
        class="flex s-MMN3rRE-BmEv"
      >
        <div
          class="w-100 s-JZthjkXlZrq3"
        >
          <label
            class="label screenreader-only s-JZthjkXlZrq3"
            for=""
          >
            
          </label>
           
          <input
            class="input s-JZthjkXlZrq3"
            id=""
            placeholder="Search..."
            type="text"
          />
           
        </div>
        <!--&lt;Input&gt;-->
         
      </div>
      <!--&lt;Search&gt;-->
    </div>
     
    <p>
      No projects available
    </p>
    
    <!--&lt;+page&gt;-->
  </div>
</body>
`;
